sopel/modules/remind.py

```diff
@@ -226,47 +226,227 @@ def remind_in(bot, trigger):
+                'year',
+                'month',
+                'day',
+                'message',
+            ]
+        )
+
+    def get_duration(self, today=None):
+        """Get the duration between the reminder and ``today``
+
+        :param today: aware datetime to compare to, default to UTC now
+        :type today: aware :class:`datetime.datetime`
+        :return: The duration, in second, between ``today`` and the reminder.
+        :rtype: :class:`int`
+
+        This method returns the number of seconds given by the
+        :class:`datetime.timedelta` obtained by the difference between the
+        reminder and ``today``.
+
+        If the delta between the reminder and ``today`` is negative, Python
+        will represent it as a negative number of days, and a positive number
+        of seconds: since it returns the number of seconds, any past reminder
+        will be transformed into a future reminder the next day.
+        """
+        if not today:
+            today = datetime.now(self.timezone)
+        else:
+            today = today.astimezone(self.timezone)
+
+        year = self.year if self.year is not None else today.year
+        month = self.month if self.month is not None else today.month
+        day = self.day if self.day is not None else today.day
 
+        at_time = datetime(
+            year, month, day,
+            self.hour, self.minute, self.second,
+            tzinfo=today.tzinfo)
 
-REGEX_AT = re.compile(r'(\d+):(\d+)(?::(\d+))?([^\s\d]+)? (.*)')
+        timediff = at_time - today
+        duration = timediff.seconds
+
+        if timediff.days > 0:
+            duration = duration + (86400 * timediff.days)
+
+        return duration
+
+
+def parse_regex_match(match, default_timezone=None):
+    """Parse a time reminder from ``match``
+
+    :param match: :obj:`~.REGEX_AT`'s matching result
+    :param default_timezone: Default trigger's timezone
+                             (from the nick's, channel's, or config's timezone)
+    :rtype: :class:`TimeReminder`
+    """
+    try:
+        timezone = validate_timezone(match.group('tz') or 'UTC')
+    except ValueError:
+        timezone = default_timezone or 'UTC'
+
+    return TimeReminder(
+        int(match.group('hours')),
+        int(match.group('minutes')),
+        int(match.group('seconds') or '0'),
+        timezone,
+        match.group('date1'),
+        match.group('date2'),
+        match.group('date3'),
+        match.group('message')
+    )
 
 
 @module.commands('at')
-@module.example('.at 13:47 Do your homework!')
+@module.example('.at 13:47 Do your homework!', user_help=True)
+@module.example('.at 03:14:07 2038-01-19 End of signed 32-bit int timestamp',
+                user_help=True)
+@module.example('.at 00:01 25/12 Open your gift!', user_help=True)
 def remind_at(bot, trigger):
-    """
-    Gives you a reminder at the given time. Takes `hh:mm:ssTimezone message`.
+    """Gives you a reminder at the given times.
```

Comment:
Who let nemesi**s** in here?!

```suggestion
    """Gives you a reminder at the given time.
```

(I mean, at least it's not a [dog](https://www.youtube.com/watch?v=KYgK97aioNA).)